Yvand.EntraCP/Yvand.EntraClaimsProvider/EntraIDEntityProvider.cs

```diff
@@ -450,6 +460,44 @@ protected virtual async Task<List<DirectoryObject>> QueryEntraIDTenantAsync(Oper
                         groupsRequestId = await batchRequestContent.AddBatchRequestStepAsync(groupRequest).ConfigureAwait(false);
                     }
 
+                    // List of groups that users must be member of, to be returned to SharePoint
+                    string[] groupsWhichUsersMustBeMemberOfRequestIds = null;
+                    string groupsWhichUsersMustBeMemberOfAny = this.Settings.GroupsWhichUsersMustBeMemberOfAny;
+                    //groupsWhichUsersMustBeMemberOfAny = "c9a94341-89b5-4109-a501-2a14027b5bf0"; // testEntraCPGroup_005 - everyone member
+                    //groupsWhichUsersMustBeMemberOfAny = "cd5f135c-9fe5-4ec2-90d9-114e9ad2e236"; // testEntraCPGroup_004 - testEntraCPUser_001 and testEntraCPUser_010 members
+                    if (!String.IsNullOrWhiteSpace(groupsWhichUsersMustBeMemberOfAny) && cachedTenantData.UserIdsMembersOfAnyRequiredGroup == null)
+                    {
+                        await cachedTenantData.WriteDataLock.WaitAsync().ConfigureAwait(false);
+                        lockToWriteInCachedDataWasTaken = true;
+                        if (cachedTenantData.UserIdsMembersOfAnyRequiredGroup != null)
```

Comment:
You just checked in the line 468 that this is null.

Reply:
Yes, here is the logic:
1. Check if UserIdsMembersOfAnyRequiredGroup is null
2. If so, wait for the lock, that is taken to fill UserIdsMembersOfAnyRequiredGroup
3. Once lock is acquired, check if, in the meantime, UserIdsMembersOfAnyRequiredGroup was not already filled by another thread. If so, no need to fill it UserIdsMembersOfAnyRequiredGroup again

Reply:
Ah, right.